### PR TITLE
Added quiz creating,updating and deleting feature

### PR DESCRIPTION
**Pull request by sanjitsarkar**

### CODE REVIEW

File: src/utils/cloudinaryService.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          body: formData,
+        });
+        response = await response.json();
+
+        imageUrls.push(response.secure_url);
+      })
+    );
+    return imageUrls;
+  } catch (err) {
+    console.log(err.message);
```
Comment (itsmebipul00) [inline src/utils/cloudinaryService.js:28]: Suggestionnn: this could be handled

**Review by itsmebipul00 — APPROVED** [review]
Good to merge

File: src/utils/cloudinaryService.js (hunk 1 above)
Comment (bossmonkey98) [reply src/utils/cloudinaryService.js:28]: +1

**Review by bossmonkey98 — APPROVED** [review]

File: src/App.css [hunk 2]
```diff
@@ -24,19 +50,30 @@
...
 }
+.modal-content{
+  padding: 0;
+  margin: 3rem auto;
+}
+.light .modal-content
+{
+  background: var(--white-color);
+}
 .light button {
```
Comment (MonikaJhingan) [inline src/App.css:89]: Use of var for color and consistent use of rem is there which is good.

File: src/components/AddQuizForm.jsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            required
+            onChange={(e) =>
+              setQuizInfo({ ...quizInfo, quizDifficulty: e.target.value })
+            }
+          >
+            <option value="" className="cursor-pointer">
+              Select quiz difficulty
+            </option>
+            <option value="Easy" className="cursor-pointer">
+              Easy
```
Comment (MonikaJhingan) [inline src/components/AddQuizForm.jsx:112]: The divsion of quiz in easy medium and hard category is good.

File: src/components/AddQuizForm.jsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            }}
+          />
+        </div>
+      </div>
+      <button className="btn btn-dark w-full text-lg mb-2" disabled={loading}>
+        {loading ? <Loader isButtonLoader={true} /> : "Add"}
+      </button>
+    </form>
+  );
+};
```
Comment (MonikaJhingan) [inline src/components/AddQuizForm.jsx:162]: The code is well indented.

File: src/components/Question.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      wrongQuestions.findIndex(
+        (wrongQuestion) => wrongQuestion.optionId === wrongOptionId
+      ) !== -1
+    );
+  };
+  return (
+    <div className="col gap-1">
+      {type === "user" && (
+        <Modal showModal={showModal} toggleModal={setShowModal}>
+          <UpdateQuestionForm question={question} toggleModal={toggleModal} />
```
Comment (MonikaJhingan) [inline src/components/Question.jsx:21]: Use of ToggleModal () is well implemented.

File: src/context/QuizContext.jsx [hunk 6]
```diff
@@ -63,48 +64,51 @@ const QuizProvider = ({ children }) => {
...
-    dispatchQuizzes({ type: ACTION_TYPE_LOADING });
+    dispatchYourQuizzes({ type: ACTION_TYPE_LOADING });
     try {
       await callApi("delete", `quizzes/${quizId}`, false);
-      dispatchQuizzes({
+      notify("Quiz deleted successfully");
+      dispatchYourQuizzes({
         type: ACTION_TYPE_SUCCESS,
-        payload: quizzes.payload.filter((quiz) => quiz.id !== quizId),
+        payload: yourQuizzes.data.filter((quiz) => quiz._id !== quizId),
```
Comment (MonikaJhingan) [inline src/context/QuizContext.jsx:90]: Use of async await is done nicely.

File: src/utils/cloudinaryService.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+export const uploadImages = async (mediaURLs, id) => {
+  try {
```
Comment (MonikaJhingan) [inline src/utils/cloudinaryService.js:2]: Use of try catch is done very well to handle the exceptions.

**Review by MonikaJhingan — APPROVED** [review]
The code and folder structure is really good, ot should be merged.

**Pull request merged**